rca/shortcourses/models.py

```diff
@@ -54,4 +76,70 @@ class ShortCoursePage(BasePage):
             heading=_("Course Introduction"),
         ),
         StreamFieldPanel("about"),
+        FieldPanel("access_planit_course_id"),
+        MultiFieldPanel(
+            [
+                FieldPanel("course_details_text"),
+                SnippetChooserPanel("frequently_asked_questions"),
+                SnippetChooserPanel("terms_and_conditions"),
+            ],
+            heading="course details",
+        ),
     ]
+
+    def get_access_planit_data(self):
+        access_planit_course_data = AccessPlanitXML(
+            course_id=self.access_planit_course_id
+        )
+        return access_planit_course_data.get_data()
+
+    def _format_booking_bar(self, register_interest_link, access_planit_data):
+        """ Booking bar messaging with the next course data available
+        Find the next course date marked as status='available' and advertise
+        this date in the booking bar. If there are no courses available, add
+        a default message."""
+
+        booking_bar = {
+            "message": "Applications are now closed",
+            "action": "Register your interest for upcoming dates",
+        }
+        # If there are no dates the booking link should go to a form, not open
+        # a modal, this link is also used as a generic interest link too though.
+        booking_bar["link"] = register_interest_link
+
+        if access_planit_data:
+            for date in access_planit_data:
+                if date["status"] == "Available":
+                    booking_bar["message"] = "Next course starts"
+                    booking_bar["date"] = date["start_date"]
+                    booking_bar["action"] = f"Book now from \xA3{date['cost']}"
+                    booking_bar["link"] = None
+                    booking_bar["modal"] = "booking-details"
+                    break
+        return booking_bar
+
+    def clean(self):
+        errors = defaultdict(list)
+        try:
+            int(self.access_planit_course_id)
```

Comment:
If `access_planit_course_id` has to be an `int`, could the field be made an integer?

Reply:
done this, thanks

Reply:
see caveat above about whether course ids could start with zero

Reply:
https://github.com/torchbox/rca-wagtail-2019/pull/335#pullrequestreview-393006124